packages/venia-concept/src/components/ToastContainer/toast.js

```diff
@@ -0,0 +1,82 @@
+import React, { useCallback } from 'react';
+import { bool, func, number, oneOf, string } from 'prop-types';
+import defaultClasses from './toast.css';
+import { useToastActions } from '@magento/peregrine';
+import { mergeClasses } from 'src/classify';
+import Icon from 'src/components/Icon';
+
+const Toast = props => {
+    const {
+        actionText,
+        dismissable,
+        icon,
+        id,
+        message,
+        onAction,
+        onDismiss,
+        type
+    } = props;
+    const classes = mergeClasses(defaultClasses, {});
+    const { removeToast } = useToastActions();
+
+    let handleDismiss, handleAction;
+    if (dismissable) {
+        handleDismiss = useCallback(() => {
```

Comment:
Should these be the responsibility of the display (this Venia component) or the data layer (within the hook/reducer)? 

Personally I think the dismiss callback can be defined within the data since it will always be the same. The question becomes whether or not we think people will want to have their toasts removed immediately when the action text is clicked or whether they want it to do something else (and then be removed? or maybe stay?).

Reply:
I like the idea of allowing for an async callback. I'll experiment with this.